docs/src/components/table-of-contents.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  colorsBackgroundUtility,
+  colorsBackgroundLight
+} from '@pluralsight/ps-design-system-core'
+import { Heading } from '@pluralsight/ps-design-system-text'
+import Theme, { useTheme } from '@pluralsight/ps-design-system-theme'
+import VerticalTabs from '@pluralsight/ps-design-system-verticaltabs'
+import React from 'react'
+import GithubSlugger from 'github-slugger'
+
+import * as styles from './table-of-contents.module.css'
+
+export interface Heading {
+  value: string
+  depth: number
+}
+
+interface TableOfContentsProps {
+  headings: Heading[]
+}
+export const TableOfContents: React.FC<TableOfContentsProps> = props => {
+  const themeName = useTheme()
+  let slugger = new GithubSlugger()
+  React.useEffect(() => {
+    slugger = new GithubSlugger()
+  }, [slugger])
```

Comment:
you could use a ref for this but i'm not sure it matters

Reply:
Seems to work as-is.  I avoid refs if I can.